x-pack/plugins/fleet/server/services/epm/archive/validation.ts

```diff
@@ -202,40 +244,87 @@ export function parseAndVerifyDataStreams(
 
     const {
       title: dataStreamTitle,
-      release,
+      release = DEFAULT_RELEASE_VALUE,
       type,
       dataset,
-      ingest_pipeline: ingestPipeline,
       streams: manifestStreams,
+      elasticsearch,
       ...restOfProps
     } = manifest;
     if (!(dataStreamTitle && type)) {
       throw new PackageInvalidArchiveError(
         `Invalid manifest for data stream '${dataStreamPath}': one or more fields missing of 'title', 'type'`
       );
     }
+
+    let ingestPipeline;
+    const ingestPipelinePaths = paths.filter((path) =>
+      path.startsWith(`${pkgKey}/data_stream/${dataStreamPath}/elasticsearch/ingest_pipeline`)
+    );
+
+    if (
+      ingestPipelinePaths.length &&
+      (ingestPipelinePaths.some((ingestPipelinePath) =>
+        ingestPipelinePath.endsWith(DEFAULT_INGEST_PIPELINE_FILE_NAME_YML)
+      ) ||
+        ingestPipelinePaths.some((ingestPipelinePath) =>
+          ingestPipelinePath.endsWith(DEFAULT_INGEST_PIPELINE_FILE_NAME_JSON)
+        ))
+    ) {
+      ingestPipeline = DEFAULT_INGEST_PIPELINE_VALUE;
+    }
+
     const streams = parseAndVerifyStreams(manifestStreams, dataStreamPath);
 
+    const parsedElasticsearchEntry: Record<string, any> = {};
+
+    if (ingestPipeline) {
+      parsedElasticsearchEntry['ingest_pipeline.name'] = DEFAULT_INGEST_PIPELINE_VALUE;
+    }
+
+    if (elasticsearch?.privileges) {
+      parsedElasticsearchEntry.privileges = elasticsearch.privileges;
+    }
+
+    if (elasticsearch?.index_template?.mappings) {
+      parsedElasticsearchEntry['index_template.mappings'] = expandDottedEntries(
+        elasticsearch.index_template.mappings
+      );
+    }
+
+    if (elasticsearch?.index_template?.settings) {
+      parsedElasticsearchEntry['index_template.settings'] = expandDottedEntries(
+        elasticsearch.index_template.settings
+      );
+    }
+
+    // Build up the stream object here so we can conditionally insert nullable fields. The package registry omits undefined
```

Comment:
A lot of the changes here are related to this concept. The registry service omits undefined values. We could probably get away with not doing this and more loosely comparing values, treating an explicit `undefined` the same as an omitted value, but I opted to be as one-to-one as possible here.

Reply:
Makes sense to be conservative for now. When we're done with this PR, could we add notes to #115032 about potential any cleanup like this?